selftests/all/unit/avocado/tree_unittest.py

```diff
@@ -165,6 +166,21 @@ def test_advanced_yaml(self):
                          oldroot.children[1].children[2].value)
         self.assertEqual({'new_value': 'something'},
                          oldroot.children[3].children[0].children[0].value)
+        # multiplex root (always True)
+        self.assertEqual(tree2.multiplex, True)
+        # multiplex /virt/
+        self.assertEqual(tree2.children[0].multiplex, True)
+        # multiplex /virt/hw
+        self.assertEqual(tree2.children[0].children[0].multiplex, True)
+        # multiplex /virt/distro
+        self.assertEqual(tree2.children[0].children[1].multiplex, False)
+        # multiplex /virt/env
+        self.assertEqual(tree2.children[0].children[2].multiplex, False)
+        # multiplex /virt/absolutly
+        self.assertEqual(tree2.children[0].children[3].multiplex, True)
+        # multiplex /virt/distro/fedora
+        self.assertEqual(tree2.children[0].children[1].children[0].multiplex,
+                         True)
 
 
 class TestPathParent(unittest.TestCase):
```

Comment:
OK, I see. So the tree structure has a multiplex field now, which doesn't look right to me. AFAIK the tree module was, up to this commit, a implementation of a tree data structure generated from YAML, without any multiplex references.

If it's specific to the multiplex implementation, then it's probably just a matter of refactoring the modules to reflect this (moving it from core/tree.py to multiplexer/tree.py or multiplexer-tree.py, or just unifying the code inside multiplexer.py)